src/presentation/http/policies/userCanReadFileData.ts

```diff
@@ -0,0 +1,46 @@
+import hasProperty from '@infrastructure/utils/hasProperty.js';
+import type { PolicyContext } from '../types/PolicyContext.js';
+import { isEmpty, notEmpty } from '@infrastructure/utils/empty.js';
+import type { NoteInternalId } from '@domain/entities/note.js';
+import { FileTypes } from '@domain/entities/file.js';
+
+/**
+ * Policy to check does user have permission to access file data,
+ * if file is a part of note
+ *
+ * @param context - Context, object containing Fatify request, Fastify reply and domain services
+ */
+export default async function userCanReadFileData(context: PolicyContext): Promise<void> {
+  const { request, reply, domainServices } = context;
+  const { userId } = request;
+
+  let noteId: NoteInternalId | null = null;
+
+  /**
+   * Get note id by file key if file is a part of note
+   */
+  if (hasProperty(request.params, 'key') && notEmpty(request.params.key)) {
+    noteId = await domainServices.fileUploaderService.getNoteIdByFileKeyAndType(request.params.key as string, FileTypes.noteAttachment);
+  } else {
+    return await reply.notAcceptable('Key not provided');
+  }
+
+  /**
+   * If note id is not resolved, we have no need to check permissions
+   */
+  if (notEmpty(noteId)) {
```

Comment:
Maybe you should use only one function? `notEmpty` or `isEmpty`

Reply:
I do not see any advantages in this

Reply:
Why two functions when you can use one?

Reply:
Because in another case, I need to make empty returns, for example. Or write '!isEmpty(..)'. Use both methods is more readable